scripts/common/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+        let project_root = find_parent_project_root(None)?;
+
+        Ok(match self {
+            PrivateKeySource::Literal(private_key) => private_key,
+            PrivateKeySource::FilePath(private_key_path) => {
+                let pk_path = make_absolute_relative_to(private_key_path, project_root)?;
+                read_secret_from_file(pk_path)?
+            }
+        })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NetworkConfig {
+    /// RPC URL
+    pub rpc_url: String,
+
+    /// Private key or path to one
+    #[serde(flatten)]
+    pub private_key_source: PrivateKeySource,
+
+    /// Additional variables
+    #[serde(flatten)]
+    pub additional_variables: BTreeMap<String, String>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct StylusConfig {
+    /// Networks' configs
+    pub networks: BTreeMap<String, NetworkConfig>,
+}
+
+#[derive(Debug)]
+pub struct Config {
+    pub client: Arc<SignerMiddleware<Provider<Http>, Wallet<SigningKey>>>,
+    pub additional_variables: BTreeMap<String, String>,
+}
+
+pub async fn load_stylus_config() -> Result<StylusConfig> {
+    let project_root = find_parent_project_root(None)?;
+    let stylus_config_path = project_root.join(STYLUS_CONFIG_FILENAME);
+    if !fs::try_exists(&stylus_config_path).await? {
+        bail!(
+            "{} not found at {}",
+            STYLUS_CONFIG_FILENAME,
+            project_root.display()
+        );
+    }
+    let stylus_config_str = fs::read_to_string(&stylus_config_path).await?;
+    Ok(toml::from_str(&stylus_config_str)?)
+}
+
+pub async fn load_network_config_for(network: &str) -> Result<NetworkConfig> {
+    let mut stylus_config: StylusConfig = load_stylus_config().await?;
+
+    stylus_config
+        .networks
+        // NOTE: `.remove`-ing instead of `.get`-ing to avoid `.clone()`-ing later
+        .remove(&network.to_lowercase())
+        .ok_or_eyre(format!("No configuration for network {}", network))
+}
+
+pub async fn load_config_for(network: &str) -> Result<Config> {
+    let NetworkConfig {
+        rpc_url,
+        private_key_source,
+        additional_variables,
+    } = load_network_config_for(network).await?;
+
+    let private_key = private_key_source.load()?;
+
+    // Prepare client
+    let provider = Provider::<Http>::try_from(rpc_url.clone())?;
+    let wallet = LocalWallet::from_str(&private_key)?;
+    let chain_id = provider.get_chainid().await?.as_u64();
+    let client = Arc::new(SignerMiddleware::new(
+        provider,
+        wallet.clone().with_chain_id(chain_id),
+    ));
+
+    Ok(Config {
+        client,
+        additional_variables,
+    })
+}
+
+/// Reads and trims a line from a filepath
+pub fn read_secret_from_file(fpath: impl AsRef<Path>) -> eyre::Result<String> {
+    let f = std::fs::File::open(fpath)?;
+    let mut buf_reader = BufReader::new(f);
+    let mut secret = String::new();
+    buf_reader.read_line(&mut secret)?;
+    Ok(secret.trim().to_string())
+}
```

Comment:
The name of the function is `read_secret_from_file` however, it can be used to read any line from any file. I suggest to rename is to `read_line_from_file` or something like that.